### PR TITLE
[Example] Recommend NFS over copy files

### PR DESCRIPTION
**Pull request by Lyken17**
## Description
Current files are synchronized via manually copying, which is hard to use (users need to specify the path mapping) and time consuming (each change on codes requires a new round of copy and there is no increamental diff). A shared network filesystem is a better way to take the role and this has been used widely in many distributed DL tasks. 

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: examples/pytorch/rgcn/experimental/README.md [hunk 1]
```diff
@@ -1,62 +1,64 @@
 ## Distributed training
 
-This is an example of training RGCN node classification in a distributed fashion. Currently, the example train RGCN graphs with input node features. The current implementation follows ../rgcn/entity_claasify_mp.py.
+This is an example of training RGCN node classification in a distributed fashion. Currently, the example only support training RGCN graphs with no input features. The current implementation follows [../rgcn/entity_claasify_mp.py](../rgcn/entity_claasify_mp.py).
```
Comment (zheng-da) [inline examples/pytorch/rgcn/experimental/README.md:3]: it supports node features now.

File: examples/pytorch/rgcn/experimental/README.md [hunk 2]
```diff
@@ -1,62 +1,64 @@
...
 
-Before training, please install some python libs by pip:
+Before training, install python libs by pip:
 
 ```bash
-sudo pip3 install ogb
-sudo pip3 install pyinstrument
-sudo pip3 install pyarrow
+pip3 install ogb pyinstrument
```
Comment (zheng-da) [inline examples/pytorch/rgcn/experimental/README.md:8]: we need to install pyarrow.

File: examples/pytorch/rgcn/experimental/README.md [hunk 3]
```diff
@@ -1,62 +1,64 @@
...
+sudo apt-get install nfs-kernel-server
+``` 
 
-Users need to make sure that the master node (node-0) has right permission to ssh to all the other nodes.
+Then create a folder for sharing and (optional) remove all restrictions
+```bash
+sudo mkdir -p /NFS
+# Change ownership and remove restrictions
+sudo chown -R nobody:nogroup /NFS
+sudo chmod 755 /NFS
```
Comment (zheng-da) [inline examples/pytorch/rgcn/experimental/README.md:28]: does this require root permission to write `/NFS`? I tried it and didn't find a way to write data to the folder with a regular account.

File: examples/pytorch/rgcn/experimental/README.md [hunk 4]
```diff
@@ -1,62 +1,64 @@
...
 
-In this example, we partition the ogbn-mag graph into 4 parts with Metis. The partitions are balanced with respect to
-the number of nodes, the number of edges and the number of labelled nodes.
+The server's internal ip can be checked  via `ifconfig` or `ip`. If the ip does not begin with `192.168`, then you may use
 ```bash
-python3 partition_graph.py --dataset ogbn-mag --num_parts 4 --balance_train --balance_edges
+# for ip range 10.0.0.0 – 10.255.255.255	
+/NFS  10.0.0.0/8(rw,sync,no_subtree_check)
+# for ip range 172.16.0.0 – 172.31.255.255	
+/NFS  172.16.0.0/12(rw,sync,no_subtree_check)
```
Comment (zheng-da) [inline examples/pytorch/rgcn/experimental/README.md:44]: after modifying /etc/exports, do we need to restart the server?

we need to provide instructions of mounting NFS from the client.

File: examples/pytorch/rgcn/experimental/README.md [hunk 5]
```diff
@@ -1,62 +1,64 @@
...
-DGL provides a script for copying partitioned data to the cluster. Before that, copy the training script to a local folder:
+
+DGL provides a script for copying partitioned data to the cluster. 
+Before that, copy the training script to a local folder:
 
 
 ```bash
 mkdir ~/dgl_code
-cp ~/dgl/examples/pytorch/rgcn/experimental/entity_classify_dist.py ~/dgl_code
+cp /home/ubuntu/dgl/examples/pytorch/rgcn/experimental/entity_classify_dist.py ~/dgl_code
```
Comment (zheng-da) [inline examples/pytorch/rgcn/experimental/README.md:59]: why do you change it to use `/home/ubuntu`?

File: examples/pytorch/rgcn/experimental/README.md [hunk 6]
```diff
@@ -75,90 +77,55 @@ python ~/dgl/tools/copy_files.py --ip_config ip_config.txt \
...
+"python3 dgl_code/entity_classify_dist.py --graph-name ogbn-mag --dataset ogbn-mag --fanout='25,25' --batch-size 512  --n-hidden 64 --lr 0.01 --eval-batch-size 16  --low-mem --dropout 0.5 --use-self-loop --n-bases 2 --n-epochs 3 --layer-norm --ip-config ip_config.txt  --num-workers 4 --num-servers 1 --sparse-embedding  --sparse-lr 0.06 --node-feats"
 ```
 
-### Step 5: Verify the partition result (Optional)
-
-```bash
-python3 verify_mag_partitions.py 
+We can get the performance score at the second epoch:
+```
+Val Acc 0.4323, Test Acc 0.4255, time: 128.0379
```
Comment (zheng-da) [inline examples/pytorch/rgcn/experimental/README.md:128]: the original README shows how to partition with ParMETIS. can you copy the instructions of using ParMETIS?

File: examples/pytorch/rgcn/experimental/README.md (hunk 1 above)
Comment (Lyken17) [reply examples/pytorch/rgcn/experimental/README.md:3]: Updated

File: examples/pytorch/rgcn/experimental/README.md (hunk 3 above)
Comment (Lyken17) [reply examples/pytorch/rgcn/experimental/README.md:28]: Usually NFS setup is performed by an admin account, so I assume that the user has sudo access. 

It is possible to use a common directory, says a folder under user's home, as a NFS point (without sudo access). However, in this case, they will not be able to edit `/etc/exports` and `/etc/fstab` to enable NFS.

File: examples/pytorch/rgcn/experimental/README.md (hunk 5 above)
Comment (Lyken17) [reply examples/pytorch/rgcn/experimental/README.md]: it is a typo. Fixed

File: examples/pytorch/rgcn/experimental/README.md (hunk 6 above)
Comment (Lyken17) [reply examples/pytorch/rgcn/experimental/README.md]: Added

File: examples/pytorch/rgcn/experimental/README.md (hunk 3 above)
Comment (zheng-da) [reply examples/pytorch/rgcn/experimental/README.md:29]: NFS setup should be done by admin. but when using NFS (e.g., writing data to NFS), we need to allow a normal user account. otherwise, it'll be very inconvenient.

we should show an example of using a folder under user's home directory. NFS setup can be done by admin. It's fine.

File: examples/pytorch/rgcn/experimental/README.md [hunk 7]
```diff
@@ -75,15 +111,41 @@ python ~/dgl/tools/copy_files.py --ip_config ip_config.txt \
...
+```
+
 ### Step 3: Launch distributed jobs
 
 DGL provides a script to launch the training job in the cluster. `part_config` and `ip_config`
 specify relative paths to the path of the workspace.
 
-The command below launches one training process on each machine and each training process has 4 sampling processes.
-**Note**: There is a known bug in Python 3.8. The training process hangs when running multiple sampling processes for each training process.
-Please set the number of sampling processes to 0 if you are using Python 3.8.
```
Comment (zheng-da) [inline examples/pytorch/rgcn/experimental/README.md:85]: we have to notify users of how to set up the sampling processes correctly. we'll fix the bug in the next release.

**Review by zheng-da — APPROVED** [review]
